src/SprykerEco/Zed/Payone/Business/Reader/PayonePaymentReader.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Payone\Business\Reader;
+
+use Generated\Shared\Transfer\OrderTransfer;
+use Generated\Shared\Transfer\PaymentDetailTransfer;
+use Generated\Shared\Transfer\PayonePaymentTransfer;
+use SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface;
+
+class PayonePaymentReader implements PayonePaymentReaderInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface
+     */
+    protected $queryContainer;
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface $queryContainer
+     */
+    public function __construct(PayoneQueryContainerInterface $queryContainer)
+    {
+        $this->queryContainer = $queryContainer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\PayonePaymentTransfer
+     */
+    public function getPayment(OrderTransfer $orderTransfer): PayonePaymentTransfer
```

Comment:
* Could you please extract this method to a repository `getPayonePaymentForOrder()` and substitute the usage of the `PayonePaymentReader` class.

Reply:
++